frontend/packages/topology/src/behavior/useDragNode.tsx

```diff
@@ -116,14 +118,14 @@ export const useDragNode = <
             }
           }
 
-          elementRef.current
-            .getController()
-            .fireEvent(
+          action(() => {
+            controller.fireEvent(
               DRAG_NODE_END_EVENT,
               elementRef.current,
               monitor.getDragEvent(),
               monitor.getOperation(),
             );
+          })();
```

Comment:
Could you add a comment here too regarding this is a workaround to async actions.